### PR TITLE
(#1780) Adds sitewide search to docs site

### LINKED ISSUE #1780: Site wide search on the NCIDS Documentation Site
In a 7/9 discussion with Anna, she requested that we develop an approach to design and build site wide search functionality into the NCIDS documentation site. https://designsystem-dev.cancer.gov/develop/

Per OCPL guidance (that we enforce across DOC sites) - every NCI owned site and application is required to have search capabilities.

This ticket is to capture that request; we'll need to determine an approach for indexing documentation site content as well as the interaction of search + search results on the documentation site.

This functionality is not required by go live of the documentation site, targeting 7/12.

~~The design system web pages should not appear in the NCI-wide search on cancer.gov.~~ It was decided 7/10/2024 that the design system site is a public site and should appear in NCI-wide search.

### Requirements
**Functional Requirements**
- The documentation search function shall only search through the NCIDS Documentation Site
- The results from the documentation search function shall only display results from content existing on the NCIDS Documentation Site

**Visual Requirements**
- The documentation search function shall appear to the users like the 
- The documentation search shall utilize NCIDS/USWDS styling

### Specifications

- The documentation search function will work like the PCP site
   - Layout template: https://github.com/NCIOCPL/gatsby-pcp/blob/develop/src/pages/search.js
- Use the site wide search app module

### DoR Tasks

- [ ] Create a PR that brings in the site wide search app module onto the Documentation Site without styling
- [ ] Review the PR with the unstyled app module and results with the Designers (@blilianyu and @monika-jaeger)
- [ ] Acceptance Criteria Drafted
- [ ] Acceptance Criteria Reviewed
  - [ ] Developer Review
  - [ ] IA Review
- [ ] Acceptance Criteria Approved
  - [ ] Developer Approval
  - [ ] IA Approval

### DoD Tasks

- [ ] PR Review (see PR for review and comments)
- [ ] Client Review with Demo

### PR DESCRIPTION
**Pull request by dlescarbeau**
**Testing this locally requires running `DEV_SSR=true pnpm start`

## Pull Request Details

_Adds sitewide search app module to docs site, including search results page and searchbar in header_

Closes #1780 

## Author PR Checklist
Items that the author of the PR is responsible for checking before submitted the PR.

### General:
- [x] I have reviewed the acceptance criteria defined in the ticket and ensured the work has been completed.
- [x] The commit message passes [all quality commit message standards](https://github.com/NCIOCPL/ncids/wiki/Creating-Quality-Commit-Messages).
- [ ] Unit tests have been updated or created to reflect any javascript changes.
- [ ] Storybook scenarios have been updated or created to reflect any html/css/js changes.

### Accessibility:
- [ ] WCAG 2.1 Level AA requirements have been met.

### Development:
- [ ] Any new or updated javascript code has 100% unit test coverage.
- [ ] New or updated breakpoints have regression images.
- [ ] Breaking changes have been thoroughly documented in the PR.

## Product Reviewer PR Checklist
Items the product team is responsible for reviewing.

### General:
- [ ] There are no unexpected or unapproved regression image changes. 
- [ ] Functionality of interactive elements meet the acceptance criteria.
- [ ] The product is visually and functionally the same across the different browsers.

### Accessibility:
- [x] AxeDev Tools: there are no new or outstanding accessibility issues introduced in this PR.
- [ ] Lighthouse: scores have not noticeably decreased during this PR.
- [ ] Wave: there are no new errors or contrast errors introduced in this PR.

## Design Reviewer PR Checklist
Items the design team is responsible for reviewing.  

### General:
- [x] New or updated features introduced in this PR are developed mobile-first.
- [x] Breakpoint changes and regression images match those breakpoints.
- [x] This PR has been tested in all supported browsers at all breakpoints.

## Developer Reviewer PR Checklist
Items the development team is responsible for reviewing.

### General:
- [x] New code passes code quality standards set by industry standards.
- [ ] The expected Storybook stories have been added or updated for the new or updated feature.
- [ ] The expected unit tests have been added or updated for the new or updated feature.

### Accessibility:
- [x] VoiceOver: Described content matches with what was expected.
- [x] Keyboard navigation: new or updated features and content are navigable via the keyboard.

### CODE REVIEW

**Review by olitharp-nci — CHANGES_REQUESTED** [review]
add the pr review template

**Comment by adrianacastaneda** [comment]
@dlescarbeau Results pull browser titles instead of the page titles. Wondering if it could pull the page title instead since all the browser titles mentioning NCIDS can be a little repetitive and distracting for users.

Seeing that some meta descriptions are being pulled so that's great. I'll make sure to get those meta descriptions in for those that don't have them!

Thank you!

cc: @bennettcc

**Comment by bryanpizzillo** [comment]
@adrianacastaneda browser titles are what the crawler finds. So, no we cannot pull in the page titles. What we probably should do is actually have `- NCI` at the end of the browser title, or add nothing at all. `- NCI` is actually the standard for browser titles and the search engine removes that from the titles it indexes.

**Comment by monika-jaeger** [comment]
@dlescarbeau looks good so far! A few things:
- Can we change the H1 "Search Results," H3 "Results for: _____," and H4 "Results ## - ## of ##"  at the top and "Results ## - ## of ##" at the bottom to match our heading styles like on c.gov using Poppins? 
- Are we able to move the pagination component at the top on the same line as the H4 so there isn't extra white space before the results? 
- The "Show ## results per page" looks like it doesn't have space around the pager select, can we add a space before and after that to give some breathing room?

cc: @bennettcc

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Comment by dlescarbeau** [comment]
> * Can we change the H1 "Search Results," H3 "Results for: _____," and H4 "Results ## - ## of ##"  at the top and "Results ## - ## of ##" at the bottom to match our heading styles like on c.gov using Poppins?
> * Are we able to move the pagination component at the top on the same line as the H4 so there isn't extra white space before the results?
> * The "Show ## results per page" looks like it doesn't have space around the pager select, can we add a space before and after that to give some breathing room?

@monika-jaeger this entire page comes from the search app module, so layout things I'll have a hard time changing, but styles I should be able to change. This means specifically:

- Yes I should be able to update the headings with our fonts.
- I can't move the pagination component to the same line of the heading unfortunately, I fear the wizardry that would take with CSS. I will try but this one is unlikely.
- Yes, I can add some padding to that select box to add some space.

File: docs/src/scss/search-page.scss [hunk 1]
```diff
@@ -0,0 +1,47 @@
+@use "uswds-core" as *;
+@use "usa-pagination" as *;
```
[the remark this answers is not in the document]
Comment (bryanpizzillo) [reply docs/src/scss/search-page.scss]: @olitharp-nci I asked Derek to do this. This way we can use as much USWDS styling as possible for our non-ncids app. This should just be making the app's classes extend the `usa-pagination` classes so we save on css.

**Comment by andyvanavery31** [comment]
Noting the differences between digital platform site search and the introduction of this site search:
- The h1 size is different on the digital platform compared to this because the h1 on this PR uses the the h1 heading style from NCIDS and that is expected.
- The pagination style on the digital platform compared to this PR is different. The style on this PR is updated to match the current USWDS style from USWDS version 3.6.0.

Requested changes:
- Can you change the margin that is above the Results For: from 16.160px to 40px so it matches the style in the digital platform?

@dlescarbeau

File: docs/src/scss/search-page.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlescarbeau) [reply docs/src/scss/search-page.scss]: @bryanpizzillo Should I put these back to the way it was? I'd changed it per Olivia's feedback.

**Comment by bryanpizzillo** [comment]
This all predates Gatsby 5. PR #1829 completed this.

**Pull request closed without merging**